### PR TITLE
EVAKA-HOTFIX Fix failing tests that create preschool applications

### PR DESCRIPTION
**Pull request by akheron**
#### Summary

E2E: Try to make sure that an active preschool term exists by moving the preferred start date to September if it's summer currently.

Integration: Change application validation to be parametrized with the current date.

### CODE REVIEW

**Review by Joosakur — APPROVED** [review]

**Review by kimifoud — APPROVED** [review]

File: frontend/src/e2e-test-common/dev-api/fixtures.ts [hunk 1]
```diff
@@ -473,76 +473,83 @@ const applicationForm = (
...
-    lastName: child.lastName,
-    socialSecurityNumber: child.ssn,
-    address: {
-      street: child.streetAddress,
-      postalCode: child.postalCode,
-      city: child.postOffice,
-      editable: false
+): ApplicationForm => {
+  let startDate = new Date()
+  if (type === 'PRESCHOOL' && [5, 6, 7, 8].includes(getMonth(startDate))) {
```
Comment (Joosakur) [inline frontend/src/e2e-test-common/dev-api/fixtures.ts:478]: `getMonth(startDate) + 1`

Comment (kimifoud) [reply frontend/src/e2e-test-common/dev-api/fixtures.ts:478]: and `setMonth(10)`

**Comment by kimifoud** [comment]
Service tests are failing too.

File: frontend/src/e2e-test-common/dev-api/fixtures.ts [hunk 2]
```diff
@@ -473,76 +473,83 @@ const applicationForm = (
...
-    address: {
-      street: child.streetAddress,
-      postalCode: child.postalCode,
-      city: child.postOffice,
-      editable: false
+): ApplicationForm => {
+  let startDate = new Date()
+  if (type === 'PRESCHOOL' && [5, 6, 7, 8].includes(getMonth(startDate) + 1)) {
+    // Try to make sure there's an active preschool term for the preferred start date
+    startDate = setMonth(startDate, 9)
```
Comment (kimifoud) [inline frontend/src/e2e-test-common/dev-api/fixtures.ts:480]: `setMonth(.., 9)` results in October. Not that it matters, but the commit message is conflicting.

Comment (akheron) [reply frontend/src/e2e-test-common/dev-api/fixtures.ts:480]: Le fu-

**Review by Joosakur — APPROVED** [review]

**Comment by Joosakur** [comment]
Still some issue with e2e test expecting a different date

**Pull request merged**